modules/upgrade/upgrade.go

```diff
@@ -8,429 +8,288 @@ import (
 	"os/exec"
 	"path/filepath"
 	"strings"
+	"syscall"
 	"time"
 
 	"github.com/pkg/errors"
 
 	"github.com/threefoldtech/zosv2/modules/zinit"
 
-	"github.com/blang/semver"
 	"github.com/rs/zerolog/log"
 
 	"github.com/threefoldtech/zosv2/modules"
 )
 
 type hookType string
 
-const (
-	hookPreCopy   hookType = "pre-copy"
-	hookPostCopy  hookType = "post-copy"
-	hookMigrate   hookType = "migrate"
-	hookPostStart hookType = "post-start"
-)
-
 const (
 	provisionModuleName = "provisiond"
-)
-
-// Upgrader is the component that is responsible
-// to keep 0-OS up to date
-type Upgrader struct {
-	root    string
-	version semver.Version
-	flister modules.Flister
-	zinit   *zinit.Client
-}
 
-// New creates a new UpgradeModule object
-func New(root string, flister modules.Flister, zinit *zinit.Client) *Upgrader {
-	if err := os.MkdirAll(root, 0770); err != nil {
-		return nil
-	}
+	// those values must match the values
+	// in the bootstrap process. (bootstrap.sh)
 
-	version, err := ensureVersionFile(root)
-	if err != nil {
-		return nil
-	}
+	// verFile the current version
+	verFile = "/tmp/version"
+	// bootFile the boot flist name
+	bootFile = "/tmp/boot"
+)
 
-	log.Info().Msgf("current version %s", version.String())
-	return &Upgrader{
-		version: version,
-		root:    root,
-		flister: flister,
-		zinit:   zinit,
-	}
-}
+var (
+	// services that can't be updated with normal procedure
+	protected = []string{"upgraded", "redis"}
+)
 
-func ensureVersionFile(root string) (version semver.Version, err error) {
-	versionPath := filepath.Join(root, "version")
-	version, err = readVersion(versionPath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			log.Error().Err(err).Msg("read version")
-			return version, err
-		}
-		log.Info().Msg("no version found, assuming fresh install")
-		// the file doesn't exist yet. So we are on a fresh system
-		version = semver.MustParse("0.0.1")
-		if err := writeVersion(versionPath, version); err != nil {
-			log.Error().Err(err).Msg("fail to write version")
-			return version, err
-		}
-	}
-	return version, nil
+//BootInfo reflects node boot information (flist + version)
+type BootInfo struct {
+	FList   string
+	Version string
 }
 
-func (u *Upgrader) enterSelfUpgrade(version semver.Version) error {
-	path := filepath.Join(u.root, "selfupgraded")
-	if err := ioutil.WriteFile(path, []byte(version.String()), 0400); err != nil {
-		return errors.Wrap(err, "fail to write selfupgrade file")
-	}
-	return nil
+// Commit write version to version file
+func (b *BootInfo) Commit(version string) error {
+	b.Version = version
+	return ioutil.WriteFile(verFile, []byte(version), 0400)
 }
 
-func (u *Upgrader) isInSelfUpgrade(version semver.Version) (bool, error) {
-	path := filepath.Join(u.root, "selfupgraded")
-
-	defer os.Remove(path)
+// DefaultBootInfo get boot info set by bootstrap process
+func DefaultBootInfo() BootInfo {
+	flist, _ := ioutil.ReadFile(bootFile)
```

Comment:
why no error checking ?

Reply:
Because there is a case where this file is not set at all (booting with overlay) so the flist string is just empty. The daemon in that case will not perform any update.